### PR TITLE
doc: Clarify tainting of nodes

### PR DESCRIPTION
**Pull request by kshlm**
This updates the README to make it clear when to use taints.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,13 +77,37 @@ Run `make ocs-registry` to generate the registry bundle container image.
...
-  To label the nodes from CLI,
+```console
+$ oc label nodes <NodeName> cluster.ocs.openshift.io/openshift-storage=''
+```
+
+OCS requires at least 3 labelled nodes.
+
+When creating StorageCluster from the UI, the create wizard takes care of labelling the selected nodes.
+
+OCS requires at least 3 nodes labelled this way.
```
Comment (obnoxxx) [inline README.md]: This is sort of duplicate. See 4 lines above.

File: README.md [hunk 2]
```diff
@@ -77,13 +77,37 @@ Run `make ocs-registry` to generate the registry bundle container image.
...
+### Dedicated nodes
+
+In case dedicated storage nodes are available, these can also be tainted to allow only OCS components to be scheduled on them.
+Nodes need to be tainted with `node.ocs.openshift.io/storage=true:NoSchedule` which can done from the CLI as follows,
+
+```console
+$ oc adm taint nodes <NodeNames> node.ocs.openshift.io/storage=true:NoSchedule
+```
+
+> Note: Not all OCS created resources have their tolerations set currently.
```
Comment (obnoxxx) [inline README.md]: I am still not really happy with this paragraph. I would put something like:

> Note: The dedicated (tainted) storage nodes will only be used by a few of the OCS components, primarily the ceph-related storage pods, like OSDs etc. These nodes won't host any other OCS components or even any apps. Therefore, _if_ you taint, you always need to have additional worker nodes, labelled for OCS, that are _not_ tainted (and optionally worker nodes that are neither labelled nor tainted). If this is not the case, i.e. if all worker nodes are tainted, the deployment will fail, and no additional apps will be able to run. [...]

File: README.md [hunk 3]
```diff
@@ -77,13 +77,37 @@ Run `make ocs-registry` to generate the registry bundle container image.
...
+> When working with a minimum sized cluster with 3 worker nodes, tainting all 3
+> nodes will cause failure of OCS to fully deploy.
+> As a workaround, an extra worker node which is not tainted with the above
+> taint can be created.
```
Comment (obnoxxx) [inline README.md]: As explained above, I would not call this a workaround. It's just a requirement if we do taint.

File: README.md [hunk 4]
```diff
@@ -77,13 +77,37 @@ Run `make ocs-registry` to generate the registry bundle container image.
...
+> StorageCluster creation using the UI wizard currently taints selected nodes
+> unconditionally. So it is required to have an additional worker node and it
+> should not be selected in the StorageCluster creation wizard.
```
Comment (obnoxxx) [inline README.md]: So this is currently a brokenness in the UI? It's OK to have it mentioned here, but it should be temporary and fixed in the UI, right?

Comment (jarrpa) [reply README.md]: There should be no mention of a "creation wizard" in this README. That is only part of the downstream product and not a part of this repo.

Comment (obnoxxx) [reply README.md]: I agree. let's remove it. (and it should only be temporary anyway)

**Comment by obnoxxx** [comment]
@kshlm while not super urgent, just a few updates are needed here. :-)

**Comment by kshlm** [comment]
@obnoxxx I've updated this.

File: README.md [hunk 5]
```diff
@@ -77,13 +77,29 @@ Run `make ocs-registry` to generate the registry bundle container image.
...
+Nodes need to be tainted with `node.ocs.openshift.io/storage=true:NoSchedule` which can done from the CLI as follows,
+
+```console
+$ oc adm taint nodes <NodeNames> node.ocs.openshift.io/storage=true:NoSchedule
+```
 
-  ```console
-  $ oc label nodes <NodeNames> cluster.ocs.openshift.io/openshift-storage=''
-  ```
+> Note: The dedicated/tainted nodes will only run OCS components. The nodes will not run any apps. Therefore, _if_ you taint, you need to have additional worker nodes that are untainted. If you don't deployment, you will be unable to run any other apps in you Openshift cluster.
```
[the remark this answers is not in the document]
Comment (obnoxxx) [reply README.md]: What about the comment that currently, we even need a non-tainted node for successful deployment as not all our components tolerate the taints?

**Pull request merged**